### PR TITLE
[tests] increase child timeout

### PR DESCRIPTION
**Pull request by bukepo**
This avoids sending data request every 10ms(`kMinPollPeriod`), which sometimes causes socket error: Operation not permitted.

### CODE REVIEW

File: src/core/thread/data_poll_manager.cpp [hunk 1]
```diff
@@ -422,7 +422,7 @@ void DataPollManager::HandlePollTimer(Timer &aTimer)
 uint32_t DataPollManager::GetDefaultPollPeriod(void) const
 {
     return TimerMilli::SecToMsec(GetNetif().GetMle().GetTimeout()) -
-           static_cast<uint32_t>(kRetxPollPeriod) * kMaxPollRetxAttempts;
+           static_cast<uint32_t>(kRetxPollPeriod) * Mle::kFailedChildTransmissions;
```
Comment (librasungirl) [inline src/core/thread/data_poll_manager.cpp]: Good catch!
Well I would prefer to update `kMaxPollRetxAttempts` to be the same as `kMaxChildKeepAliveAttempts`(i.e. 4) as 

1. it is where `kMniTimeout` derives from `kMinTimeout = (((kMaxChildKeepAliveAttempts + 1) * kUnicastRetransmissionDelay) / 1000), ///< Minimum timeout(s)`. And

2. keep data_poll_manager less dependent on constants defined in other files.

Thoughts?

File: tests/scripts/thread-cert/Cert_5_1_02_ChildAddressTimeout.py [hunk 2]
```diff
@@ -65,13 +65,13 @@ def setUp(self):
 
         self.nodes[ED].set_panid(0xface)
         self.nodes[ED].set_mode('rsn')
-        self.nodes[ED].set_timeout(3)
+        self.nodes[ED].set_timeout(6)
```
Comment (librasungirl) [inline tests/scripts/thread-cert/Cert_5_1_02_ChildAddressTimeout.py]: can we use a constant or macro for the value in a common file?

File: src/core/thread/data_poll_manager.cpp (hunk 1 above)
Comment (bukepo) [reply src/core/thread/data_poll_manager.cpp]: Defined configuration in `openthread-core-default-config.h` and updated the expression to calculate kMniTimeout.

File: src/core/thread/mle_constants.hpp [hunk 3]
```diff
@@ -74,7 +74,13 @@ enum
     kMaxChildIdRequestTimeout       = 5000,  ///< Maximum delay for receiving a Child ID Request
     kMaxChildUpdateResponseTimeout  = 2000,  ///< Maximum delay for receiving a Child Update Response
     kMaxLinkRequestTimeout          = 2000,  ///< Maximum delay for receiving a Link Accept
-    kMinTimeout = (((kMaxChildKeepAliveAttempts + 1) * kUnicastRetransmissionDelay) / 1000), ///< Minimum timeout(s)
+    kMinTimeoutKeepAlive            = (((kMaxChildKeepAliveAttempts + 1) * kUnicastRetransmissionDelay) /
+                            1000), ///< Minimum timeout(s) for keep alive
+    kMinTimeoutDataPoll             = (OPENTHREAD_CONFIG_MINIMUM_POLL_PERIOD +
+                           OPENTHREAD_CONFIG_FAILED_CHILD_TRANSMISSIONS * OPENTHREAD_CONFIG_RETX_POLL_PERIOD) /
+                          1000, ///< Minimum timeout(s) for data poll
```
Comment (librasungirl) [inline src/core/thread/mle_constants.hpp]: `OPENTHREAD_CONFIG_MINIMUM_POLL_PERIOD` here might be pointless if `/1000`

**Pull request merged**